### ISSUE TITLE
Изменение поведения программы

### ISSUE THREAD

**Issue by xtemp09** [Aegel5/SimpleSwitcher#56]
Спасибо за такую программу, после всех мытарств я был рад перейти на неё.

Я использовал _Punto Switcher_ до тех пор, пока Яндекс не стал добавлять к программе размером 3,2 Мб свой клон хрома и установщик не стал занимать 30 Мб. Я продолжил использовать старую версию _Punto_, затем пришлось перейти на _Caramba Switcher_, но, к сожалению, он не настраивается, так что был просто вынужден его удалить после недолгого использования.
Эта программа позволила набирать текст почти как в старом _Punto Switcher_!

Можно ли попросить многоуважаемого создателя программы ввести режим, при котором при нажатии горячей клавиши, набранный или выделенный текст менялся из одной раскладки в другую?

Дело в том, что сейчас этого нет, текст удаляется и перенабирается без оглядки на него.

ФдЗЩ становится aLPO, а если при этом Caps Lock отжат, aLpo (должен быть AlPO).
192,168,0,1 становится 192?168?0?1 (должен стать 192.168.0.1)

Как _Punto_, так и _Caramba_ меняли текст из одной раскладки в другую.

**Comment by Aegel5** [comment]
Добрый день. 

"ФдЗЩ становится aLPO, а если при этом Caps Lock отжат, aLpo" - эта ошибка у меня не воспроизводится. Похоже у вас какие-то проблемы с Capslock. На какую клавишу у вас настроено исправление?

"192,168,0,1 становится 192?168?0?1" - если запятая была набрана через Shift на русской раскладке, то логично что теперь здесь ? который набирается этой же клавишей на английской раскладке.

**Comment by xtemp09** [comment]
> Похоже у вас какие-то проблемы с Capslock. На какую клавишу у вас настроено исправление?

Pause Break

Я сейчас в Word 2019 набрал ФдЗЩ, нажал Pause Break, получил aLPO.

У меня на Num Pad нарисованы "," и "Del". В русской раскладке клавиша выдаёт ",", в английской ― ".", поэтому, когда я нажимаю Pause Break, я ожидаю замену запятых на точки, а получаю замену запятых на знаки вопроса.

Если я набираю не на Num Pad, в английской раскладке "192/168/0/1", выделяю фрагмент, нажимаю Shift + Pause Break, получается "192|168|0|1". На клавише у меня нарисованы "?", ",", "/" и ".".

Клавиатура Logitech K270, раскладка "Английский (США) - США", Windows 10 Pro 21H2, 19044.3086.

**Comment by Aegel5** [comment]
>У меня на Num Pad нарисованы "," и "Del". В русской раскладке клавиша выдаёт ",", в английской ― ".", поэтому, когда я нажимаю Pause Break, я ожидаю замену запятых на точки, а получаю замену запятых на знаки вопроса.

Попробовал - ошибка не повторяется. Тут есть один момент. Когда исправление происходит через выделение текста, программа не знает, в какой раскладке она набрана и исправляет "по умолчанию". А вот если исправлять сразу после набора - все работает ok у меня. Если у вас не правильно исправляет сразу после набора (при наборе через num pad), то нужны логи с повторением ошибки.

**Comment by Aegel5** [comment]
> Если я набираю не на Num Pad, в английской раскладке "192/168/0/1", выделяю фрагмент, нажимаю Shift + Pause Break, получается "192|168|0|1". На клавише у меня нарисованы "?", ",", "/" и ".".

Если во время исправления выделенного текста стоит русская раскладка, то клавише / соответствует символ | в английской раскладке (на моей клавиатуре это клавиша справа от э).

**Comment by Aegel5** [comment]
>Я сейчас в Word 2019 набрал ФдЗЩ, нажал Pause Break, получил aLPO.

А в других программах (например wordpad) повторяется?

**Comment by xtemp09** [comment]
Вот лог. [SimpleSwitcher.exe(5664)_7150921.log](https://github.com/Aegel5/SimpleSwitcher/files/14110948/SimpleSwitcher.exe.5664._7150921.log)



Ввёл ФдЗЩ, отжал Caps Lock, нажал Pause Break, стало Alpo.

Ввёл AlPO, отжал капслок, нажал Pause Break, получилось фДЗЩ.

**Comment by xtemp09** [comment]
> А в других программах (например wordpad) повторяется?

Да.

**Comment by Aegel5** [comment]
> Ввёл ФдЗЩ, отжал Caps Lock, нажал Pause Break, стало Alpo.

Перед вводом капслок был включен или нет?

**Comment by xtemp09** [comment]
> Перед вводом капслок был включен или нет?

Нет, только для ввода двух последних символов.

**Comment by Aegel5** [comment]
> Нет, только для ввода двух последних символов.

Да, теперь у меня повторилось. Действительно, переключение CapsLock во время набора текста ломает исправление.

**Comment by AnrDaemon** [comment]
Отключите CapsLock. Или ещё лучше - повесьте на него переключение языка. Решите сразу две проблемы.
